src/ng/compile.js

```diff
@@ -3101,6 +3130,27 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
         }
       });
 
+      function recordChanges(key, currentValue, previousValue) {
+        if (isFunction(destination.$onChanges)) {
+          // If we have not already scheduled the onChanges hook then do so now
+          if (!changes) {
+            changes = {};
+            scope.$$postDigest(triggerOnChangesHook);
+          }
+          // Store this change
+          changes[key] = {previousValue: previousValue, currentValue: currentValue};
+        }
+      }
+
+      function triggerOnChangesHook() {
+        // We must run this hook in an apply since the $$postDigest runs outside apply
+        scope.$apply(function() {
```

Comment:
Doesn't this mean there will be one extra digest for each controller (that has at least one changed binding) per digest ?